tests/gradients/test_finite_difference.py

```diff
@@ -140,15 +140,67 @@ def test_independent_parameter(self, mocker):
         # only called for parameter 0
         assert spy.call_args[0][0:2] == (tape, 0)
 
-    @pytest.mark.parametrize("interface", ["autograd", "jax", "torch", "tensorflow"])
-    def test_no_trainable_params_qnode(self, interface):
+    @pytest.mark.autograd
+    def test_no_trainable_params_qnode_autograd(self):
         """Test that the correct ouput and warning is generated in the absence of any trainable
         parameters"""
-        if interface != "autograd":
-            pytest.importorskip(interface)
         dev = qml.device("default.qubit", wires=2)
 
-        @qml.qnode(dev, interface=interface)
+        @qml.qnode(dev, interface="autograd")
+        def circuit(weights):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=0)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        weights = [0.1, 0.2]
+        with pytest.warns(UserWarning, match="gradient of a QNode with no trainable parameters"):
+            res = qml.gradients.finite_diff(circuit)(weights)
+
+        assert res == ()
+
+    @pytest.mark.torch
+    def test_no_trainable_params_qnode_torch(self):
+        """Test that the correct ouput and warning is generated in the absence of any trainable
+        parameters"""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(weights):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=0)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        weights = [0.1, 0.2]
+        with pytest.warns(UserWarning, match="gradient of a QNode with no trainable parameters"):
+            res = qml.gradients.finite_diff(circuit)(weights)
+
+        assert res == ()

```

Comment:
We could move all of this to a class helper method `def check_no_trainable_params(self, interface)`.

Then have individual interface tests like: 


```
    @pytest.mark.tf
    def test_no_trainable_params_qnode_tf(self):
        self.check_no_trainable_params("tensorflow")
```

Or we could use markers in conjunction with `pytest.mark.parametrize`, see [the docs](https://docs.pytest.org/en/stable/example/markers.html#marking-individual-tests-when-using-parametrize)

```
@pytest.mark.parametrize("interface", (pytest.param("autograd", marks=pytest.mark.autograd),
pytest.param("jax", marks=pytest.mark.jax), pytest.param("torch", marks=pytest.mark.torch), pytest.param("tensorflow", marks=pytest.mark.tf)) )
```

Reply:
I'd be willing to implement this is relevant places.